air/src/runner/outcome.rs

```diff
@@ -59,6 +60,7 @@ pub(crate) fn from_success_result(exec_ctx: ExecutionCtx<'_>, trace_handler: Tra
 /// set ret_code based on the error.
 pub(crate) fn from_preparation_error(data: impl Into<Vec<u8>>, err: PreparationError) -> InterpreterOutcome {
     let ret_code = err.to_error_code() as i32;
+    let ret_code = PREPARATION_ERRORS_START_ID + ret_code;
```

Comment:
Maybe move that into `to_error_code` implementation?

Regarding `PREPARATION_ERRORS_START_ID`, maybe it's not needed? This seems like a very general approach: all error codes must start with `1`, not just `PreparationError`. And I don't think `1` will be changed to anything else, will it?

Reply:
I think it's possible to implement `to_error_code` for [IntoEnumIterator](https://docs.rs/strum/0.22.0/strum/trait.IntoEnumIterator.html) trait. For example, `ExecutionError` also uses same approach in `to_error_code`, it would makes sense to start from `1` there, too.

Reply:
ExecutionErrors starts from `EXECUTION_ERRORS_START_ID = 100`, it's intended to distinguish between them

Reply:
regarding moving that into `to_error_code` - it makes sense, thanks